packages/core/src/build/configAndIndexingFunctions.ts

```diff
@@ -360,8 +442,38 @@ export async function buildConfig({
         }
 
         // TODO: Explicit validation of indexed argument value format (array or object).
+        // The first element of the array return from `buildTopics` being defined
+        // is an invariant of the current filter design.
         // Note: This can throw.
-        topics = buildTopics(rawContract.abi, rawContract.filter);
+        const [topic0FromFilter, ...topicsFromFilter] = buildTopics(
+          rawContract.abi,
+          rawContract.filter,
+        ) as [Exclude<LogTopic, null>, ...LogTopic[]];
```

Comment:
I'm still not sure I follow this. Isn't this line converting the user-provided `filter` object into a `topics` value? I feel like instead of asserting the type, we should have a runtime validation here to confirm that the `topic0` value returned from `buildTopics` is of shape `0xabc...` or [`0xabc...`, `0xdef...`].

Look a few lines down, I think this line could get moved out of the loop:
```ts
const filteredEventSelectors = Array.isArray(topic0FromFilter)
  ? topic0FromFilter
  : [topic0FromFilter];
```

And then, we could validate what I described above at runtime. This would validate against a filter that resolves to an empty `topic0`, eg:
```ts
contract: {
  filter: {
    event: []
  }
}
```

Reply:
The filter that the user provides can only result in topic0 that has the shape `Hex | Hex[]`. This is because we force the user to provide an `event` to the filter, which gets directly translated into topic0.